src/interfaces/plugins.ts

```diff
@@ -601,11 +581,11 @@ module.exports = (theApp: any) => {
     }
 
     try {
-      const pluginConstructor: (
-        app: ServerAPI
-        // eslint-disable-next-line @typescript-eslint/no-var-requires
-      ) => PluginInfo = require(path.join(location, packageName))
-      plugin = pluginConstructor(appCopy)
+      const moduleDir = path.join(location, packageName)
+      const pluginConstructor: PluginConstructor = await importOrRequire(
+        moduleDir
+      )
+      plugin = pluginConstructor(appCopy) as PluginInfo
```

Comment:
Time to move the props in PluginInfo to Plugin and remove PluginInfo?

Reply:
I moved the properties that are considered part of the plugin interface into `server-api`, but the properties remaining on `PluginInfo` get added by the server while loading the plugin and aren't part of the public interface.

Reply:
Right, that part makes total sense.

This cast just seems fishy: cast to a type with a bunch of non-optional properties that we know are not in the object from the constructor.

Can we move the property setters here and cast after the props are there?